pkg/templates/compile.go

```diff
@@ -119,7 +119,11 @@ func Parse(filePath string, preprocessor Preprocessor, options protocols.Executo
 // templates.
 //
 // TODO: support all protocols.
-func (template *Template) isGlobalMatchersEnabled() bool {
+func (template *Template) isGlobalMatchersEnabled(executorOpts protocols.ExecutorOptions) bool {
+	if !executorOpts.Options.EnableGlobalMatchersTemplates {
```

Comment:
afair, `template.Options ` is available. If so, we can use it instead of passing the options.